pac4j-core/src/main/java/org/pac4j/core/profile/service/InMemoryProfileService.java

```diff
@@ -70,18 +71,18 @@ protected void deleteById(final String id) {
 	@Override
 	protected List<Map<String, Object>> read(final List<String> names, final String key, final String value) {
 		logger.debug("Reading key / value: {} / {}", key, value);
-		final List<Map<String, Object>> listAttributes = new ArrayList<>();
+		final List<Map<String, Object>> listAttributes;
 		if (key.equals(getIdAttribute())) {
+			listAttributes = new ArrayList<>();
 			final Map<String,Object> profile = profiles.get(value);
```

Comment:
Well, first, I really need to ask: what is your motivation to use the Stream API in general (performance? readability?)? Also, does it have to be used everywhere? I can't help having the feeling that it's just a new toy... It can be useful in some places, but if it was the holy grail it would have been there for a long time...

Also, the reason why I think Stream is useless here, is that what I'm doing here is a `.get(myId)`, I'm not walking through the list. I can walk through the list and make a filter to check if the key is `myId`, but that would mean replacing the very efficient `.get` by something completely inefficient. And it would be, in my opinion, less readable. So, what I'm perceiving here is that it would harm performance and readability, but would use the new shiny toy... and I'm not willing to do that.

Please correct me and tell me why the stream api should be used here.